app/test/testmerginapi.cpp

```diff
@@ -2942,3 +2942,31 @@ void TestMerginApi::testParseVersion()
   QCOMPARE( major, 2024 );
   QCOMPARE( minor, 4 );
 }
+
+void TestMerginApi::testUpdateProjectMetadataRole()
+{
+  QString projectName = "testUpdateProjectMetadataRole";
+
+  createRemoteProject( mApiExtra, mWorkspaceName, projectName, mTestDataPath + "/" + TEST_PROJECT_NAME + "/" );
+  downloadRemoteProject( mApi, mWorkspaceName, projectName );
+
+  LocalProject projectInfo = mApi->localProjectsManager().projectFromMerginName( mWorkspaceName, projectName );
+  QVERIFY( projectInfo.isValid() );
+  QString projectDir = projectInfo.projectDir;
+
+  // Test 1: Initial role should be 'owner'
+  QString cachedRole = mApi->getCachedProjectRole( MerginApi::getFullProjectName( mWorkspaceName, projectName ) );
+  QCOMPARE( cachedRole, QString( "owner" ) );
+
+  // Test 2: Update cached role to 'reader'
+  QString newRole = "reader";
+  bool updateSuccess = mApi->updateCachedProjectRole( MerginApi::getFullProjectName( mWorkspaceName, projectName ), newRole );
+  QVERIFY( updateSuccess );
+
+  // Verify role was updated in cache
+  cachedRole = mApi->getCachedProjectRole( MerginApi::getFullProjectName( mWorkspaceName, projectName ) );
+  QCOMPARE( cachedRole, newRole );
```

Comment:
This is fine, but we now need to test the actual behaviour - communication with the server. Let's call the `reloadProjectRole` method and check if the cached role will be back to `owner` as it should :)

Reply:
Just a tip - it is handy to qcompare to real values instead of variables (if possible). This way unit tests are more readable.

```suggestion
  QCOMPARE( cachedRole, "reader" );
```